ragengine/tests/vector_store/test_faiss_store.py

```diff
@@ -0,0 +1,225 @@
+
+def test_index_documents(vector_store_manager):
+    documents = [
+        Document(doc_id="1", text="First document", metadata={"type": "text"}),
+        Document(doc_id="2", text="Second document", metadata={"type": "text"})
+    ]
+    
+    doc_ids = vector_store_manager.index_documents("test_index", documents)
+    
+    assert len(doc_ids) == 2
+    assert doc_ids == ["1", "2"]
+
+def test_index_documents_isolation(vector_store_manager):
+    doc_1_id, doc_2_id = "1", "2"
+    documents1 = [
+        Document(doc_id=doc_1_id, text="First document in index1", metadata={"type": "text"}),
+    ]
+    documents2 = [
+        Document(doc_id=doc_2_id, text="First document in index2", metadata={"type": "text"}),
+    ]
+
+    # Index documents in separate indices
+    index_name_1, index_name_2 = "index1", "index2"
+    vector_store_manager.index_documents(index_name_1, documents1)
+    vector_store_manager.index_documents(index_name_2, documents2)
+
+    # Ensure documents are correctly persisted and separated by index
+    doc_1 = vector_store_manager.get_document(index_name_1, doc_1_id)
+    assert doc_1 and doc_1.node_ids # Ensure documents were created
+
+    doc_2 = vector_store_manager.get_document(index_name_2, doc_2_id)
+    assert doc_2 and doc_2.node_ids # Ensure documents were created
+
+    # Ensure that the documents do not mix between indices
+    assert vector_store_manager.get_document(index_name_2, doc_1_id) is None, f"Document {doc_1_id} should not exist in {index_name_2}"
+    assert vector_store_manager.get_document(index_name_1, doc_2_id) is None, f"Document {doc_2_id} should not exist in {index_name_1}"
+
+@patch('requests.post')
+def test_query_documents(mock_post, vector_store_manager):
+    # Define Mock Response for Custom Inference API
+    mock_response = {
+        "result": "This is the completion from the API"
+    }
+
+    mock_post.return_value.json.return_value = mock_response
+
+    # Add documents to index
+    documents = [
+        Document(doc_id="1", text="First document", metadata={"type": "text"}),
+        Document(doc_id="2", text="Second document", metadata={"type": "text"})
+    ]
+    vector_store_manager.index_documents("test_index", documents)
+
+    params = {"temperature": 0.7}
+    # Mock query and results
+    query_result = vector_store_manager.query("test_index", "First", top_k=1, params=params)
+
+    assert query_result is not None
+    assert query_result.response == "This is the completion from the API"
+
+    mock_post.assert_called_once_with(
+        INFERENCE_URL,
+        # Auto-Generated by LlamaIndex
+        json={"prompt": "Context information is below.\n---------------------\ntype: text\n\nFirst document\n---------------------\nGiven the context information and not prior knowledge, answer the query.\nQuery: First\nAnswer: ", "formatted": True, 'temperature': 0.7},
+        headers={"Authorization": f"Bearer {INFERENCE_ACCESS_SECRET}"}
+    )
+
+"""
```

Comment:
Let us remove all the commented code. They serve no values for now.

Reply:
done